store/store.go

```diff
@@ -2089,6 +2090,23 @@ func (s *Store) SnapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 	}
 }
 
+func genInstanceKey(curSnap *CurrentSnap) string {
+	_, snapInstanceKey := snap.SplitInstanceName(curSnap.InstanceName)
+
+	if snapInstanceKey == "" {
+		return curSnap.SnapID
+	}
+
+	// due to privacy concerns, avoid sending the local names to the
+	// backend, instead hash the snap ID and instance key together
+	h := crypto.SHA512.New()
+	// TODO parallel-install: include seed
+	h.Write([]byte(curSnap.SnapID))
+	h.Write([]byte(snapInstanceKey))
+	h.Write([]byte(curSnap.RequestSeed))
+	return fmt.Sprintf("%s-%x", curSnap.SnapID, h.Sum(nil)[39:])
```

Comment:
Given that we throw away most of the sha512 result I wonder if we should just use a shorter hash like sha1. Or is there an actual attack scenario where the longer sha benefits us?

Reply:
yes, sha512 feels a bit overkill here,  sha256 or sha224 would be fine

Reply:
I would try to encode the hash a bit more compactly, url safe base64 maybe

Reply:
Switched to SHA256 + Bse64. Note that with Base64, `-` may appear int the output. One has to remember to do `foo.split('-', maxsplit=2)` at the backend side.